src/components/TaskForm.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from "react";
+
+import { DateInput } from "@mantine/dates";
+import { Container, Select, TextInput, Button, Group } from "@mantine/core";
+import "@mantine/dates/styles.css";
+
+import { Task } from "../types/Task";
+import "./styles.css";
+
+interface TaskFormProps {
+  addTask: (task: Task) => void;
+}
+
+const TaskForm: React.FC<TaskFormProps> = ({ addTask }) => {
+  const [title, setTitle] = useState<string>("");
+  const [summary, setSummary] = useState<string>("");
+  const [priority, setPriority] = useState<number>(2);
+  const [dueDate, setDueDate] = useState<Date | null>(null);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!title || !summary || !priority || !dueDate) {
+      alert("Please fill in all fields");
+      return;
+    }
+    const newTask: Task = {
+      id: new Date().toISOString(),
+      title,
+      summary,
+      priority,
+      dueDate,
+      completed: false,
+    };
+    addTask(newTask);
+    setTitle("");
+    setSummary("");
+    setPriority(2);
+    setDueDate(null);
+  };
+
+  return (
+    <Container
+      className="formContainer"
+      bg="var(--mantine-color-dark-5)"
+      h="auto"
+      mt={50}
+      pb={15}
+    >
+      <form onSubmit={handleSubmit}>
+        <Group>
+          <TextInput
+            size="md"
+            className="titleInput"
+            label="Title"
+            placeholder="Give a title for your task"
+            value={title}
+            onChange={(e) => setTitle(e.currentTarget.value)}
+            required
+          ></TextInput>
```

Comment:
```suggestion
          />
```

Can use self closing tag here.